tests/ulyanov_et_al_2016/test_modules.py

```diff
@@ -0,0 +1,143 @@
+import itertools
+from typing import Dict, cast
+
+import pytest
+
+import pytorch_testing_utils as ptu
+import torch
+from torch import nn
+
+from pystiche.image.utils import extract_num_channels
+from pystiche_papers.ulyanov_et_al_2016 import modules
+
+
+def test_SequentialWithOutChannels(subtests):
+    sequential_modules = [nn.Conv2d(3, 3, 1), nn.Conv2d(3, 3, 1)]
+    for out_channel_name in (
+        None,
+        1,
+        "1",
+    ):  # TODO: Rename out_channel_name is not implemented
+        module = modules.SequentialWithOutChannels(
+            *sequential_modules, out_channel_name=out_channel_name
+        )
+
+        with subtests.test("out_channel_name"):
+            if out_channel_name is None:
+                desired_out_channel_name = "1"
+            elif isinstance(out_channel_name, int):
+                desired_out_channel_name = str(out_channel_name)
+            else:
+                desired_out_channel_name = out_channel_name
+
+            assert (
+                tuple(cast(Dict[str, nn.Module], module._modules).keys())[-1]
+                == desired_out_channel_name
+            )
+
+        with subtests.test("out_channels"):
+            assert module.out_channels == sequential_modules[-1].out_channels
+
+
+def test_join_channelwise(subtests, input_image, style_image):
+    join_image = modules.join_channelwise(input_image, style_image)
+    assert isinstance(join_image, torch.Tensor)
+    input_num_channels = extract_num_channels(input_image)
+    with subtests.test("num_channels"):
+        assert extract_num_channels(
+            join_image
+        ) == input_num_channels + extract_num_channels(style_image)
+    with subtests.test("input_image"):
+        ptu.assert_allclose(join_image[:, :input_num_channels, :, :], input_image)
+    with subtests.test("style_image"):
+        ptu.assert_allclose(join_image[:, input_num_channels:, :, :], style_image)
+
+
+def test_UlyanovEtAl2016StylizationDownsample(subtests):
+    module = modules.UlyanovEtAl2016StylizationDownsample()
+    assert isinstance(module, nn.AvgPool2d)
+    with subtests.test("kernel_size"):
+        assert module.kernel_size == pytest.approx(2)
```

Comment:
`kernel_size` should be an `int` or a `tuple` of `int`s. Thus, `pytest.approx` is not needed here.